src/node/ext/node_grpc.cc

```diff
@@ -298,16 +325,98 @@ NAN_METHOD(SetDefaultRootsPem) {
   }
 }
 
+NAUV_WORK_CB(LogMessagesCallback) {
+  Nan::HandleScope scope;
+  std::list<gpr_log_func_args *> args;
+  uv_mutex_lock(&grpc_logger_state.mutex);
+  args.splice(args.begin(), *grpc_logger_state.pending_args);
+  uv_mutex_unlock(&grpc_logger_state.mutex);
+  /* Call the callback with each log message */
+  while (!args.empty()) {
+    gpr_log_func_args *arg = args.front();
+    args.pop_front();
+    Local<Value> file = Nan::New(arg->file).ToLocalChecked();
+    Local<Value> line = Nan::New<Uint32, uint32_t>(arg->line);
+    Local<Value> severity = Nan::New(
+        gpr_log_severity_string(arg->severity)).ToLocalChecked();
+    Local<Value> message = Nan::New(arg->message).ToLocalChecked();
+    const int argc = 4;
+    Local<Value> argv[argc] = {file, line, severity, message};
+    grpc_logger_state.callback->Call(argc, argv);
+    delete[] arg->message;
+    delete arg;
+  }
+}
+
+void node_log_func(gpr_log_func_args *args) {
+  // TODO(mlumish): Use the core's log formatter when it becomes available
+  gpr_log_func_args *args_copy = new gpr_log_func_args;
+  size_t message_len = strlen(args->message) + 1;
+  char *message = new char[message_len];
+  memcpy(message, args->message, message_len);
+  memcpy(args_copy, args, sizeof(gpr_log_func_args));
+  args_copy->message = message;
```

Comment:
Done. Now any custom logger will also include ISO timestamps in the output, but we're limited to millisecond resolution because of how Node represents dates.